Saqib_mini_project_todo

saqibjvd opened the pull request:
done week 1.. all tasks

saqibjvd committed:
todo list

delte individual task and delete all working. marked completed wip

saqibjvd committed:
week 1 task completed

completed all mandatories and optional task.. .still got to do css.

jxz12 commented on app/saqib/page.js:
```diff
@@ -3,14 +3,15 @@
+  // this function will add new task to the list
   function addTask(e) {
     e.preventDefault();
     const newTodo = {
```
the `id` here is actually causing the error, sorry I gave you the wrong logic. You can copy Saliha's logic for this!

jxz12 commented on app/saqib/page.js:
```diff
@@ -27,42 +44,50 @@ export default function TodoApp() {
+                onClick={(e) => clickHandler(todo)}
               >
                 {todo.task}
+                <button onClick={() => deleteItem(todo.id)} className="delete">
```
this does not work because the `onClick` on the parent `<li>` is taking priority over the button's `onClick` if you move the parent's `onClick` into a `<span>` like this then it should work:
```js
                <span onClick={(e) => clickHandler(todo)}>
                  {todo.task}
                </span>
```

saqibjvd committed:
week1

fix errors with id, adding task after delete all task. still having issue with marking task as completed /uncompleted

saqibjvd committed:
Update page.js

mark completed not working ...i tried with id, todo, todo.id

saqibjvd commented:
hi , my mark as completed not working

saqibjvd committed:
Update page.js
